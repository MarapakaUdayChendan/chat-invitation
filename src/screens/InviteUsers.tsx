import React, { useState } from "react";
import { View, Text, StyleSheet, TouchableOpacity, ScrollView } from "react-native";
import { COLORS, FONT, INPUT } from "../styles/theme";
import { useNavigation } from "expo-router";
import { NativeStackNavigationProp } from "@react-navigation/native-stack";
import { RootStack } from "../navigation/RootStackNavigation";

const InviteUsers: React.FC = () => {
  const [options, setOptions] = useState([
    { id: 1, label: "E-Mail", checked: false },
    { id: 2, label: "SMS", checked: false },
    { id: 3, label: "What's app", checked: false },
    { id: 4, label: "Invite through chat (Platform Users)", checked: false },
  ]);
  const navigation = useNavigation<NativeStackNavigationProp<RootStack>>();
  const toggleCheckbox = (id: number) => {
    setOptions((prev) =>
      prev.map((opt) => (opt.id === id ? { ...opt, checked: !opt.checked } : opt))
    );
  };

  const handleSubmit = () => {
    const selected = options.filter((opt) => opt.checked).map((opt) => opt.label);
    console.log("Selected Options:", selected);
    navigation.navigate('MobileScreen')
  };

  return (
    <View style={styles.container}>
      <ScrollView contentContainerStyle={styles.scrollContent}>
        {options.map((option) => (
          <TouchableOpacity
            key={option.id}
            style={styles.checkboxContainer}
            onPress={() => toggleCheckbox(option.id)}
          >
            <View style={[styles.checkbox, option.checked && styles.checkedBox]}>
              {option.checked && <View style={styles.innerCheck} />}
            </View>
            <Text style={styles.label}>{option.label}</Text>
          </TouchableOpacity>
        ))}
      </ScrollView>

      <TouchableOpacity style={styles.button} onPress={handleSubmit}>
        <Text style={styles.buttonText}>Submit</Text>
      </TouchableOpacity>
    </View>
  );
};

export default InviteUsers;

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: COLORS.background,
  },
  scrollContent: {
    padding: 16,
    paddingBottom: 100,
  },
  checkboxContainer: {
    flexDirection: "row",
    alignItems: "center",
    backgroundColor: '#add0f752',
    padding: 12,
    borderRadius: 8,
    marginBottom: 12,
  },
  checkbox: {
    width: 24,
    height: 24,
    borderWidth: 2,
    borderColor: COLORS.primary,
    borderRadius: 4,
    marginRight: 12,
    justifyContent: "center",
    alignItems: "center",
  },
  checkedBox: {
    backgroundColor: COLORS.primary,
  },
  innerCheck: {
    width: 12,
    height: 12,
    backgroundColor: COLORS.onPrimary,
    borderRadius: 2,
  },
  label: {
    fontSize: FONT.size.input,
    color: COLORS.onSurface,
    flexShrink: 1, // ensures long text wraps
  },
  button: {
    position: "absolute",
    bottom: 16,
    alignSelf: "center",
    width: "90%",
    backgroundColor: COLORS.primary,
    paddingVertical: 16,
    borderRadius: 8,
    alignItems: "center",
  },
  buttonText: {
    color: COLORS.onPrimary,
    fontSize: FONT.size.button,
    fontWeight: FONT.weight.medium,
  },
});
