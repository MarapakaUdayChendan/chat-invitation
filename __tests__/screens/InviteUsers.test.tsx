import React from "react";
import { render } from "@testing-library/react-native";
import InviteUsers from "../InviteUsers";

test("renders InviteUsers component", () => {
  const { toJSON } = render(<InviteUsers />);
  expect(toJSON()).toMatchSnapshot();
});
