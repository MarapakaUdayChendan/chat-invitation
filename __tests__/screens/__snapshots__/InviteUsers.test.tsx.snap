// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InviteUsers renders InviteUsers component and matches snapshot 1`] = `
<View
  style={
    {
      "backgroundColor": "#FFFFFF",
      "flex": 1,
    }
  }
>
  <RCTScrollView
    contentContainerStyle={
      {
        "padding": 16,
        "paddingBottom": 100,
      }
    }
  >
    <View>
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "alignItems": "center",
            "backgroundColor": "#add0f752",
            "borderRadius": 8,
            "flexDirection": "row",
            "marginBottom": 12,
            "opacity": 1,
            "padding": 12,
          }
        }
      >
        <View
          style={
            [
              {
                "alignItems": "center",
                "borderColor": "#007AFF",
                "borderRadius": 4,
                "borderWidth": 2,
                "height": 24,
                "justifyContent": "center",
                "marginRight": 12,
                "width": 24,
              },
              false,
            ]
          }
        />
        <Text
          style={
            {
              "color": "#000000",
              "flexShrink": 1,
              "fontSize": 16,
            }
          }
        >
          E-Mail
        </Text>
      </View>
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "alignItems": "center",
            "backgroundColor": "#add0f752",
            "borderRadius": 8,
            "flexDirection": "row",
            "marginBottom": 12,
            "opacity": 1,
            "padding": 12,
          }
        }
      >
        <View
          style={
            [
              {
                "alignItems": "center",
                "borderColor": "#007AFF",
                "borderRadius": 4,
                "borderWidth": 2,
                "height": 24,
                "justifyContent": "center",
                "marginRight": 12,
                "width": 24,
              },
              false,
            ]
          }
        />
        <Text
          style={
            {
              "color": "#000000",
              "flexShrink": 1,
              "fontSize": 16,
            }
          }
        >
          SMS
        </Text>
      </View>
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "alignItems": "center",
            "backgroundColor": "#add0f752",
            "borderRadius": 8,
            "flexDirection": "row",
            "marginBottom": 12,
            "opacity": 1,
            "padding": 12,
          }
        }
      >
        <View
          style={
            [
              {
                "alignItems": "center",
                "borderColor": "#007AFF",
                "borderRadius": 4,
                "borderWidth": 2,
                "height": 24,
                "justifyContent": "center",
                "marginRight": 12,
                "width": 24,
              },
              false,
            ]
          }
        />
        <Text
          style={
            {
              "color": "#000000",
              "flexShrink": 1,
              "fontSize": 16,
            }
          }
        >
          What's app
        </Text>
      </View>
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "alignItems": "center",
            "backgroundColor": "#add0f752",
            "borderRadius": 8,
            "flexDirection": "row",
            "marginBottom": 12,
            "opacity": 1,
            "padding": 12,
          }
        }
      >
        <View
          style={
            [
              {
                "alignItems": "center",
                "borderColor": "#007AFF",
                "borderRadius": 4,
                "borderWidth": 2,
                "height": 24,
                "justifyContent": "center",
                "marginRight": 12,
                "width": 24,
              },
              false,
            ]
          }
        />
        <Text
          style={
            {
              "color": "#000000",
              "flexShrink": 1,
              "fontSize": 16,
            }
          }
        >
          Invite through chat (Platform Users)
        </Text>
      </View>
    </View>
  </RCTScrollView>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "alignSelf": "center",
        "backgroundColor": "#007AFF",
        "borderRadius": 8,
        "bottom": 16,
        "opacity": 1,
        "paddingVertical": 16,
        "position": "absolute",
        "width": "90%",
      }
    }
  >
    <Text
      style={
        {
          "color": "#FFFFFF",
          "fontSize": 18,
          "fontWeight": "500",
        }
      }
    >
      Submit
    </Text>
  </View>
</View>
`;
